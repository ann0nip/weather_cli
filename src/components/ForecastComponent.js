import React from "react";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import { CardMedia, Divider, ListItem, ListItemText } from "@material-ui/core";

const useStyles = makeStyles({
  forecastIcon: {
    height: 55,
    width: 55,
    backgroundSize: 55,
  },
});

export default function ForecastComponent({ data }) {
  const {
    dt_txt,
    main: { temp_max },
    weather: [{ icon, description }],
  } = data;
  const classes = useStyles();
  return (
    <React.Fragment>
      <ListItem>
        <ListItemText primary={moment(dt_txt).format("MMM Do")} />
        <ListItemText primary={`${Math.trunc(temp_max)} CÂº`} />
        <CardMedia
          className={classes.forecastIcon}
          image={`http://openweathermap.org/img/wn/${icon}@2x.png`}
          title={description}
        />
      </ListItem>
      <Divider variant="middle" />
    </React.Fragment>
  );
}
