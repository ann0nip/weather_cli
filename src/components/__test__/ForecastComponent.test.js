import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { data } from "./ForecastComponent.mock";
import ForecastComponent from "../ForecastComponent";

import { CardMedia, ListItemText } from "@material-ui/core";

describe("<ForecastComponent />", () => {
  it("render <ForecastComponent /> component", () => {
    const tree = renderer.create(<ForecastComponent data={data} />).toJSON();
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(<ForecastComponent data={data} />);
    wrapper.find(ListItemText).forEach((item) => {
      expect(item.prop("primary")).toBeDefined();
    });
    expect(wrapper.find(CardMedia)).toHaveLength(1);
  });
});
