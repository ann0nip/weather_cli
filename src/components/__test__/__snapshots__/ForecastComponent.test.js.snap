// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastComponent /> render <ForecastComponent /> component 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Jul 1st
      </span>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        10 CÂº
      </span>
    </div>
    <div
      className="MuiCardMedia-root makeStyles-forecastIcon-1"
      style={
        Object {
          "backgroundImage": "url(\\"http://openweathermap.org/img/wn/10n@2x.png\\")",
        }
      }
      title="light rain"
    />
  </li>,
  <hr
    className="MuiDivider-root MuiDivider-middle"
  />,
]
`;
